src/main/java/org/openrewrite/java/spring/boot3/ReplaceMockBeanAndSpyBean.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot3;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.ChangeType;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+
+public class ReplaceMockBeanAndSpyBean extends Recipe {
+    private static final AnnotationMatcher MOCKBEAN_ANNOTATION_MATCHER =
+            new AnnotationMatcher("@org.springframework.boot.test.mock.mockito.MockBean");
+    private static final AnnotationMatcher SPYBEAN_ANNOTATION_MATCHER =
+            new AnnotationMatcher("@org.springframework.boot.test.mock.mockito.SpyBean");
+
+
+    @Override
+    public String getDisplayName() {
+        return "Replace @MockBean and  @SpyBean";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces `@MockBean` and `@SpyBean` annotations with `@MockitoBean` and `@MockitoSpyBean`. " +
+                "Also Update the relevant import statements.";
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+
+                J.Annotation a = super.visitAnnotation(annotation, ctx);
+
+                // Check if the annotation is @MockBean
+                if (MOCKBEAN_ANNOTATION_MATCHER.matches(a)) {
+                    //remove the old import and add the new one
+                    maybeRemoveImport("org.springframework.boot.test.mock.mockito.MockBean");
+                    maybeAddImport("org.springframework.test.context.bean.override.mockito.MockitoBean");
+
+                    //Change the annotation
+                    a = (J.Annotation) new ChangeType("org.springframework.boot.test.mock.mockito.MockBean",
+                            "org.springframework.test.context.bean.override.mockito.MockitoBean", false)
+                            .getVisitor().visit(a, ctx, getCursor().getParentOrThrow());
```

Comment:
That is a good point 👍   
I've started with a java file because that is the first thing in the rewrite docs for writing your own recipe.  
I couldn't find similar documentation for the yaml aproach. Maybe I'm blind, but to me it seems that the entry hurdle for yaml is higher for new users. Not because of the complexity, just because of missing/hard to find documentation.  

I've combed through some examples and managed to get this running with yaml.

That being said, I checked the api docs for the 2 annotations again and it seems they have some optional parameters that won't be supported anymore by the replacements.  Do we need to handle that somehow?

https://docs.spring.io/spring-boot/api/java/org/springframework/boot/test/mock/mockito/MockBean.html
https://docs.spring.io/spring-framework/docs/6.2.0/javadoc-api/org/springframework/test/context/bean/override/mockito/MockitoBean.html

https://docs.spring.io/spring-boot/api/java/org/springframework/boot/test/mock/mockito/SpyBean.html
https://docs.spring.io/spring-framework/docs/6.2.0/javadoc-api/org/springframework/test/context/bean/override/mockito/MockitoSpyBean.html

Reply:
That's good feedback, thanks! Usually we recommend folks run through [this workshop](https://docs.moderne.io/user-documentation/workshops/recipe-authoring/) that does introduce the concepts with exercises in the order we recommend, but perhaps more can be done to nudge folks that way when they land on the Java recipe authoring page. /cc @mike-solomon

As for the no-longer-supported attributes; if they can be removed without migration then we can reuse the declarative `org.openrewrite.java.RemoveAnnotationAttribute`. If there's some more complex migration then it might help to add a few tests here first to showcase the required before/after, and then we'll work out the details from there.

Reply:
Thanks for that link. I didn't see that page at all before.  
I'll make sure to work through it these days 😃 

In the meantime I think I addressed the attributes.  
fortunatly it is just one rename and otherwise removals.  
Please note that while `@MockBean` and `@MockitoBean` both have the attribute `value`, it means completely different things.   
Luckily `value` in `@MockBean` is just an alias for `classes` which we can simply remove 😃

Reply:
Awesome, thanks a lot for working that in!